src/hubs/generic/basehub.ts

```diff
@@ -1,45 +1,31 @@
 import { EventEmitter } from "events";
 
-import { IBLEAbstraction } from "../interfaces";
-
-import { ColorDistanceSensor } from "../devices/colordistancesensor";
-import { CurrentSensor } from "../devices/currentsensor";
-import { Device } from "../devices/device";
-
-import { DuploTrainBaseColorSensor } from "../devices/duplotrainbasecolorsensor";
-import { DuploTrainBaseMotor } from "../devices/duplotrainbasemotor";
-import { DuploTrainBaseSpeaker } from "../devices/duplotrainbasespeaker";
-import { DuploTrainBaseSpeedometer } from "../devices/duplotrainbasespeedometer";
-
-import { HubLED } from "../devices/hubled";
-import { Light } from "../devices/light";
-import { MediumLinearMotor } from "../devices/mediumlinearmotor";
-import { MotionSensor } from "../devices/motionsensor";
-import { MoveHubMediumLinearMotor } from "../devices/movehubmediumlinearmotor";
-import { MoveHubTiltSensor } from "../devices/movehubtiltsensor";
-import { RemoteControlButton } from "../devices/remotecontrolbutton";
-import { SimpleMediumLinearMotor } from "../devices/simplemediumlinearmotor";
-import { PiezoBuzzer } from "../devices/piezobuzzer";
-import { TechnicLargeLinearMotor } from "../devices/techniclargelinearmotor";
-import { TechnicMediumHubAccelerometerSensor } from "../devices/technicmediumhubaccelerometersensor";
-import { TechnicMediumHubGyroSensor } from "../devices/technicmediumhubgyrosensor";
-import { TechnicMediumHubTiltSensor } from "../devices/technicmediumhubtiltsensor";
-import { TechnicXLargeLinearMotor } from "../devices/technicxlargelinearmotor";
-import { TiltSensor } from "../devices/tiltsensor";
-import { TrainMotor } from "../devices/trainmotor";
-import { VoltageSensor } from "../devices/voltagesensor";
-
-import * as Consts from "../consts";
+import { IBLEAbstraction } from "../../interfaces";
+
+import { Device, DeviceVersion } from "../../devices/generic/device";
+
+import { devices, deviceTypeNames } from "../../devices";
+
+import * as Consts from "../../consts";
 
 import Debug = require("debug");
 const debug = Debug("basehub");
 
-
 /**
  * @class BaseHub
  * @extends EventEmitter
  */
 export class BaseHub extends EventEmitter {
+    public static get type () {
+        return this._type;
+    }
+    public static get typeName () {
+        return this._typeName;
+    }
+
+    protected static _type: number = 0;
+    protected static _typeName: string = "UNKNOW";
+    protected static _portMap: {[portName: string]: number} = {};
```

Comment:
Add type, type name and port map to static attributes to remove namespace and related consts